### PR TITLE
DM-33521: Split plugins out of ctrl_bps

### PR DESCRIPTION
**Pull request by mxk62**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+      - id: check-yaml
+        args:
+          - "--unsafe"
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/psf/black
+    rev: 22.1.0
```
Comment (timj) [inline .pre-commit-config.yaml:11]: Can you make this 22.3.0 please?

File: doc/changes/README.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* ``api``: An API change.
+* ``perf``: A performance enhancement.
+* ``doc``: A documentation improvement.
+* ``removal``: An API removal or deprecation.
+* ``other``: Other Changes and Additions of interest to general users.
+* ``misc``: Changes that are of minor interest.
+
+An example file name would therefore look like ``DM-30291.misc.rst``.
+
+If the change concerns specifically the registry or a datastore the news fragment can be placed in the relevant subdirectory.
```
Comment (timj) [inline doc/changes/README.rst:21]: Butler copy and paste. Remove.

File: examples/SConscript [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# -*- python -*-
+from lsst.sconsUtils import scripts
```
Comment (timj) [inline examples/SConscript:2]: You can delete the examples directory if it's empty.

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,46 @@
+[tool.towncrier]
+    package = "lsst.ctrl.bps"
```
Comment (timj) [inline pyproject.toml:2]: This needs updating, as does the filename and title_format below.

**Review by timj — APPROVED** [review]
Looks good. Only minor comments.

File: doc/lsst.ctrl.bps.panda/index.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+   userguide.rst
+
+.. _lsst.ctrl.bps.panda-contributing:
+
+Contributing
+============
+
+``lsst.ctrl.bps.panda`` is developed at https://github.com/lsst/ctrl_bps_panda.
+You can find Jira issues for this module under the `ctrl_bps_panda <https://jira.lsstcorp.org/issues/?jql=project%20%3D%20DM%20AND%20component%20%3D%20ctrl_bps_panda>`_ component.
```
Comment (MichelleGower) [inline doc/lsst.ctrl.bps.panda/index.rst:37]: Will there be this new component or just the current PanDA component?

File: doc/lsst.ctrl.bps.panda/userguide.rst [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+   https://panda.lsst.io.
+
+.. _panda-plugin-wmsclass:
+
+Specifying the plugin
+---------------------
+
+The class providing `PanDA`_ support for `ctrl_bps`_ is ::
+
+    lsst.ctrl.bps.htcondor.PanDAService
```
Comment (MichelleGower) [inline doc/lsst.ctrl.bps.panda/userguide.rst:24]: htcondor?

File: doc/lsst.ctrl.bps.panda/userguide.rst [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+Inform `ctrl_bps`_ about its location using one of the methods described in its
+`documentation`__.
+
+.. __: https://pipelines.lsst.io/v/weekly/modules/lsst.ctrl.bps/index.html
+
+ .. _htc-plugin-defining-submission:
```
Comment (MichelleGower) [inline doc/lsst.ctrl.bps.panda/userguide.rst:31]: Should this line be here?

File: doc/lsst.ctrl.bps.panda/userguide.rst [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+
+.. _panda-plugin-defining-submission:
+
+Defining a submission
+---------------------
+
+BPS configuration files are YAML files with some reserved keywords and some
+special features. See `BPS configuration file`__ for details.
+
+.. Describe any plugin specific ascpects of a definiing a submissinon below if
```
Comment (MichelleGower) [inline doc/lsst.ctrl.bps.panda/userguide.rst:41]: typos: ascpects and submissinon

File: doc/lsst.ctrl.bps.panda/userguide.rst [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+
+LSST Batch Processing Service (BPS) allow large-scale workflows to execute in
+well-managed fashion, potentially in multiple environments. The service is
+provided by `ctrl_bps`_ package. ``ctrl_bps_panda`` is a plugin allowing `ctrl_bps`_
+to execute workflows on computational resources managed by `PanDA`_.
+
+.. note::
+
+   Documentation describing how to submit workflows for execution using PanDA
+   deployment in Interim Data Facility (*aka* Google cloud) can can be found at
```
Comment (MichelleGower) [inline doc/lsst.ctrl.bps.panda/userguide.rst:14]: typo:  can can

**Review by MichelleGower — APPROVED** [review]
I ran a test workflow with this on the idf and it ran successfully.  Some minor questions/comments mostly about the documentation.

File: doc/lsst.ctrl.bps.panda/userguide.rst (hunk 6 above)
Comment (mxk62) [reply doc/lsst.ctrl.bps.panda/userguide.rst:24]: Yeah, it should read `panda`. Will fix.

File: doc/lsst.ctrl.bps.panda/userguide.rst (hunk 7 above)
Comment (mxk62) [reply doc/lsst.ctrl.bps.panda/userguide.rst:31]: Nope, will remove.

File: doc/lsst.ctrl.bps.panda/userguide.rst (hunk 8 above)
Comment (mxk62) [reply doc/lsst.ctrl.bps.panda/userguide.rst:41]: It looks like `vim`s spellchecker does not highlight spelling errors in words in comments. Will fix.

**Pull request merged**